### PR TITLE
Add daysPerMonth var in .h and removed other declaration

### LINKED ISSUE #241: daysPerMonth is defined 5 times (at least)
```bash
libs/antares/study/parts/hydro/container.cpp:151:52:                static const uint daysPerMonth[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
libs/antares/study/parts/hydro/container.cpp:289:48:            static const uint daysPerMonth[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
libs/antares/study/parts/hydro/series.cpp:151:44:        static const uint daysPerMonth[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
libs/antares/logs/cleaner.cpp:39:36:static const uint daysPerMonth[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
analyzer/atsp/atsp.cpp:37:32:const uint ATSP::lonmois[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
```

Remove multiple definitions. If possible, place a single definition/declaration in a dedicated header.

### PR DESCRIPTION
**Pull request by payetvin**
close #241

### CODE REVIEW

File: src/libs/antares/logs/cleaner.cpp [hunk 1]
```diff
@@ -36,7 +37,6 @@ namespace Antares
 {
 // days per month, immutable values for version prior to 3.9 for sure
 // these values are not related in any cases to an Antares study
```
Comment (flomnes) [inline src/libs/antares/logs/cleaner.cpp]: Remove those comments

File: src/libs/antares/study/parts/hydro/container.cpp [hunk 2]
```diff
@@ -148,7 +148,6 @@ bool PartHydro::LoadFromFolder(Study& study, const AnyString& folder)
             {
                 // days per month, immutable values for version prior to 3.9 for sure
                 // these values was hard-coded before 3.9
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/container.cpp]: Remove those comments

File: src/libs/antares/study/parts/hydro/container.cpp [hunk 3]
```diff
@@ -286,7 +285,6 @@ bool PartHydro::LoadFromFolder(Study& study, const AnyString& folder)
                   && ret;
 
             double temp[3][DAYS_PER_YEAR];
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/container.cpp:284]: `DAYS_PER_YEAR` could be defined in **constants.h** as well, just under `HOURS_PER_YEAR`

File: src/libs/antares/study/parts/hydro/series.cpp [hunk 4]
```diff
@@ -148,7 +148,6 @@ bool DataSeriesHydro::loadFromFolder(Study& study, const AreaName& areaID, const
 
         std::vector<std::vector<double>> temp(countInt, std::vector<double>(DAYS_PER_YEAR));
 
-        static const uint daysPerMonth[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
         uint daysPerMonthDecals[12];
         for (int oldMonth = 0; oldMonth < 12; oldMonth++)
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/series.cpp]: It seems that this code is duplicate (see container.cpp). Can you take a look ?

File: src/libs/antares/constants.h [hunk 5]
```diff
@@ -40,6 +40,8 @@
 /*! Hours per year */
 #define HOURS_PER_YEAR 8760
 
+static const uint daysPerMonth[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
```
Comment (flomnes) [inline src/libs/antares/constants.h]: Use an `std::array` if possible

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parts/hydro/container.cpp (hunk 3 above)
Comment (payetvin) [reply src/libs/antares/study/parts/hydro/container.cpp:284]: It's already defined there, should I move it or do something else ?

Comment (flomnes) [reply src/libs/antares/study/parts/hydro/container.cpp:284]: Sorry, ignore this

File: src/libs/antares/constants.h [hunk 6]
```diff
@@ -40,7 +41,7 @@
 /*! Hours per year */
 #define HOURS_PER_YEAR 8760
 
-static const uint daysPerMonth[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+static const std::array<uint, 12> daysPerMonth = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
```
Comment (flomnes) [inline src/libs/antares/constants.h]: This defines as many symbols as there are translation units, which can be a problem if there are many of them. Consider this
**constants.h**
```cpp
extern const std::array<uint, 12> daysPerMonth;
```
**constants.cpp**
```cpp
const std::array<uint, 12> daysPerMonth = {31, /* ..., */ 31};
```

File: src/libs/antares/study/parts/hydro/series.cpp [hunk 7]
```diff
@@ -321,5 +303,31 @@ uint64 DataSeriesHydro::memoryUsage() const
     return sizeof(double) + ror.memoryUsage() + storage.memoryUsage();
 }
 
+void DataSeriesHydro::AdjustMonth(Study& study, uint(& firstDayMonth)[13], uint(& daysPerMonthDecals)[12])
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/series.cpp]: No need for references here

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parts/hydro/series.cpp [hunk 8]
```diff
@@ -322,5 +303,31 @@ uint64 DataSeriesHydro::memoryUsage() const
     return sizeof(double) + ror.memoryUsage() + storage.memoryUsage();
 }
 
+void DataSeriesHydro::AdjustMonth(Study& study, uint firstDayMonth[13], uint daysPerMonthDecals[12])
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/series.cpp]:
```suggestion
void DataSeriesHydro::AdjustMonth(const Study& study, uint firstDayMonth[13])
```

- `study` isn't changed
- `uint daysPerMonthDecals[12]` is just a temporary variable, you can declare it within the impl of `DataSeriesHydro::AdjustMonth`

File: src/libs/antares/study/parts/hydro/series.cpp [hunk 9]
```diff
@@ -322,5 +303,31 @@ uint64 DataSeriesHydro::memoryUsage() const
...
+{
+    for (int oldMonth = 0; oldMonth < 12; oldMonth++)
+    {
+        int realMonth = (oldMonth + study.parameters.firstMonthInYear) % 12;
+        daysPerMonthDecals[oldMonth] = daysPerMonth[realMonth];
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/series.cpp]:
```suggestion
        daysPerMonthDecals[oldMonth] = Constants::daysPerMonth[realMonth];
```

File: src/libs/antares/constants.cpp [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include "constants.h"
+
+const std::array<uint, 12> daysPerMonth = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
```
Comment (flomnes) [inline src/libs/antares/constants.cpp]:
```suggestion
namespace Antares::Constants {
const std::array<uint, 12> daysPerMonth = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
}
```

File: src/libs/antares/constants.h [hunk 11]
```diff
@@ -40,6 +41,8 @@
 /*! Hours per year */
 #define HOURS_PER_YEAR 8760
 
+extern const std::array<uint, 12> daysPerMonth;
+
```
Comment (flomnes) [inline src/libs/antares/constants.h]:
```suggestion
namespace Antares::Constants {
extern const std::array<uint, 12> daysPerMonth;
}

```

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parts/hydro/series.cpp (hunk 8 above)
Comment (payetvin) [reply src/libs/antares/study/parts/hydro/series.cpp]: daysPerMonthDecals is used outside of the function, L 166 and 169

**Review by flomnes — APPROVED** [review]

**Pull request merged**